tests/components/tasmota/test_common.py

```diff
@@ -50,10 +46,17 @@ async def help_test_availability_when_connection_lost(hass, mqtt_mock, domain, c
     mqtt_mock.connected = False
     async_dispatcher_send(hass, MQTT_DISCONNECTED)
     await hass.async_block_till_done()
-
+    await hass.async_block_till_done()
     state = hass.states.get(f"{domain}.test")
     assert state.state == STATE_UNAVAILABLE
 
+    mqtt_mock.connected = True
+    async_dispatcher_send(hass, MQTT_CONNECTED)
```

Comment:
Can we generate this event without relying directly on the mqtt integration signal which is a detail?

Reply:
Not really, we could patch `mqtt.subscribe_connection_status` + `mqtt.is_connected`, is that better?

Reply:
Is `mqtt_client_mock.on_connect` the intact real `_mqtt_on_connect` callback?

Reply:
Thanks a lot, that works well.